### PR TITLE
Work around AutoRest schema validation by changing to x-az- prefix

### PR DESCRIPTION
**Pull request by tg-msft**
Updating all of our vendor extensions to use an `x-az-` prefix instead of `x-ms-` to
prevent the AutoRest schema validator from complaining.  Also fixing a couple of
other schema violations and changing from header `collectionFormat` `multi` to
`csv` (which is correct).

Please note there are **NO changes to the code** after regenerating.

### CODE REVIEW

File: sdk/storage/Azure.Storage.Common/swagger/Generator/src/generator.ts [hunk 1]
```diff
@@ -557,9 +554,9 @@ function generateOperation(w: IndentWriter, serviceModel: IServiceModel, group:
                     w.line(`if (${responseName}.Headers.TryGetValue("${header.name}", out ${headerName}))`);
                     w.scope('{', '}', () => {
                         w.write(`${valueName}.${naming.pascalCase(header.clientName)} = `);
-                        if (isPrimitiveType(header.model) && header.model.collectionFormat === `multi`) {
+                        if (isPrimitiveType(header.model) && header.model.collectionFormat === `csv`) {
```
Comment (pakrym) [inline sdk/storage/Azure.Storage.Common/swagger/Generator/src/generator.ts]: Is this related to the PR?

**Review by pakrym — APPROVED** [review]

File: sdk/storage/Azure.Storage.Common/swagger/Generator/src/generator.ts (hunk 1 above)
Comment (tg-msft) [reply sdk/storage/Azure.Storage.Common/swagger/Generator/src/generator.ts]: Yes - the part where we change from `multi` to `csv` for header params.

**Comment by tg-msft** [comment]
Heads up @seanmcc-msft and @kfarmer-msft -- I'm going to merge this around the end of the day.
